### DISCUSSION TITLE
Add listen to live audio during a pass

### DISCUSSION THREAD

**Discussion by stefi01** [jekhokie/raspberry-noaa-v2#495] [Ideas]
I think it could be cool to add live audio so we can listen to the pass as it happens without downloading the audio files.

**Comment by Cadair** [comment]
This is (obviously) technically possible, and would be a really cool feature.

I think it would be a lot of work, so someone would have to be pretty motivated to try and make it happen, things I can think of that would need doing:

* Looking into if it's possible to get gnuradio to pipe the audio out to somewhere at the correct point in the workflow
* Figuring out what the best way to stream that to a user of the web panel would be (something like a rtmp/hls stream but for audio only I would guess).

My main concern would be doing this for every pass when you might only listen to 1/10 or less, I would want to investigate how much system resources it would take up, certainly if we enabled it by default.

**Comment by dom-robinson** [comment]
Pretty sure it will be easy to adapt the gnuradio .grc sources to add an audio out for a local audio device, and possibly could add an ffmpeg (cpu intensive) that could present the audio as a single stream so a single user could listen through the web page to the capturing output. GNURadio audio out being simpler and 'first' since it shouldn't add any (much) cpu. Ill have a look once the meteordemod update is in and we are stable on both buster and bullseye.

might get to some of that this week
